middlewares/authenticate.js

```diff
@@ -0,0 +1,30 @@
+const jwt = require('jsonwebtoken');
+const { User } = require('../models/user');
+
+require('dotenv').config();
+const { SECRET_KEY } = process.env;
+
+const { HttpErorrs } = require('../helpers'); 
+const CtrlWrapper = require('../helpers/CtrlWrapper');
+
+const authenticate = async (req, res, next) => {
+    const { authorization = "" } = req.headers;
+    const [bearer, token] = authorization.split(' ');
+
+    if (bearer !== 'Bearer') {
+        throw HttpErorrs(401, 'Not authorized');
+    }
+try {
+    const { id } = jwt.verify(token, SECRET_KEY);
+    const user = await User.findById(id);
+
+    if (!user) {
```

Comment:
you should define req.user here and req.user must be equal to decrypted user from jwt token. I didn't see it here. Does it really work?

Reply:
seemed to work, but I rewrote the code